src/OrleansProviders/Streams/Common/PooledCache/PooledQueueCache.cs

```diff
@@ -97,8 +93,16 @@ public PooledQueueCache(ICacheDataAdapter<TQueueMessage, TCachedMessage> cacheDa
             this.cacheDataAdapter = cacheDataAdapter;
             this.comparer = comparer;
             this.logger = logger.GetSubLogger("messagecache", "-");
+            this.itemCount = 0;
             pool = new CachedMessagePool<TQueueMessage, TCachedMessage>(cacheDataAdapter);
             messageBlocks = new LinkedList<CachedMessageBlock<TCachedMessage>>();
+            this.cacheMonitor = cacheMonitor;
+
+            if (this.cacheMonitor != null && cacheMonitorWriteInterval.HasValue)
+            {
+                var timer = new Timer(this.ReportCacheMessageStatistics, null, cacheMonitorWriteInterval.Value, cacheMonitorWriteInterval.Value);
```

Comment:
If the timer was a member of the class like this.timer  - new Timer(..), it would be kept around until the cache went away, but we don't store a reference of it anywhere, so it will be cleaned up by garbage collection, right?  Once that happens the timer will be turned off.  Unless I'm misunderstanding how/when objects are garbage collected..